src/domain/Session.js

```
@@ -39,14 +40,18 @@ export default class Session {
   uuid: string;
   profile: ?Profile;
 
-  static parse(plain: Response): Session {
+  static parse(plain: Response | BadResponse): ?Session {
+    if (plain instanceof BadResponse) {
+      return null;
+    }
```

Je ne polluerais pas le domaine et les utilisateurs du SDK avec des `null` et des des `BadResponse`. À ce stade, il ne devrait pas avoir d'erreur et vous n'auriez pas à valider ceci. 

Votre `ApiRequester` devrait lancer un exception moindrement que ce n'est pas un 2xx. Si c'est un 4xx, c'est une exception qu'on peut prévenir généralement. Si c'est un erreur 5xx, c'est une situation qu'on _doit_ gérer anyway. Ça fait tout simplement du sens de laisser l'exception remonter à l'utilisateur du SDK. Ce n'est pas au SDK à gérer ces cas, et j'ajouterais même que vous avalez l'exception, alors on n'a aucune idée de ce qui se passe. Par exemple, j'ai vu ailleurs que vous retournez un array vide si c'est un `BadResponse`: qui me dit que ce n'est pas une situation normale de n'avoir aucune donnée? Fail fast.